openapi.yaml

```
@@ -7856,9 +7858,17 @@ components:
         account: 1
       properties:
         for:
-          description: List of repository groups for which to calculate the metrics.
-          items:
-            $ref: '#/components/schemas/RepositorySet'
+          description: |
+            List of repository groups for which to calculate the metrics.
+            An empty repository (empty array) group will match nothing.
+            Use `null` as repository group to match every repository.
+          items:
+            anyOf:
+            - $ref: '#/components/schemas/RepositorySet'
+            - items:
+                $ref: '#/components/schemas/RepositoryID'
+              nullable: true
+              type: array
```

This should work better
```suggestion
            oneOf:
            - $ref: '#/components/schemas/RepositorySet'
            - type: "null"
```

---

type: 'null' is not supported by openapi even if supported by jsonschema
https://swagger.io/docs/specification/data-models/data-types/
luckily this is working, it's simple enough:
```
            oneOf:
            - $ref: '#/components/schemas/RepositorySet'
            - enum:
              - null
```